transactions.md

```diff
@@ -3,4 +3,93 @@ title: Transactions
+
+{{site.data.alerts.callout_info}}In CockroachDB, individual statements outside of transactions are considered implicit transactions and are retried automatically; no special error handling is needed.{{site.data.alerts.end}}
+
+## Transaction Priorities
+
+Every transaction in CockroachDB is assigned an initial **priority**. By default, that priority is `NORMAL`, but for transactions that should be given preference in high-contention scenarios, the client can set the priority within the `BEGIN TRANSACTION` statement:
+
+~~~ sql
+BEGIN TRANSACTION PRIORITY <LOW, NORMAL, HIGH>
+~~~
+
+Alternately, the client can set the priority immediately after the transaction is started as follows:
+
+~~~ sql
+SET TRANSACTION PRIORITY <LOW, NORMAL, HIGH>
+~~~
+
+### Priorities and Retries
+
+When two transactions contend for the same resource, the one with the lower priority loses and is retried. On retry, the transaction inherits the priority of the winner. This means that each retry makes a transaction stronger and more likely to succeed.
+
+## Isolation Levels
+
+CockroachDB supports two transaction isolation levels: `SNAPSHOT ISOLATION` and `SERIALIZABLE`. By default, transactions use the `SERIALIZABLE` isolation level, but the client can explicitly set a transaction's isolation when starting the transaction:
+
+~~~ sql
+BEGIN TRANSACTION ISOLATION LEVEL <ANSI SQL ISOLATION LEVEL>
+~~~
+
+Alternately, the client can set the isolation level immediately after the transaction is started as follows:
+
+~~~ sql
+SET TRANSACTION ISOLATION LEVEL <ANSI SQL ISOLATION LEVEL>
+~~~
+
+### SERIALIZABLE ISOLATION
+
+With `SERIALIZABLE` isolation, a transaction behaves as though it has the entire database all to itself for the duration of its execution. This means that no concurrent writers can affect the transaction unless they commit before it starts, and no concurrent readers can be affected by the transaction until it has successfully committed. This is the strongest level of isolation provided by CockroachDB and it's the default. 
+
+Unlike `SNAPSHOT ISOLATION`, `SERIALIZABLE` isolation permits no anomalies. However, due to CockroachDB's transaction model, `SERIALIZABLE` isolation may require more transaction restarts, especially in the presence of high contention between concurrent transactions. Consider using `SNAPSHOT ISOLATION` for high concurrency workloads.
+
+### SNAPSHOT ISOLATION
+
+With `SNAPSHOT ISOLATION`, a transaction behaves as if it
+were reading the state of the database consistently at a fixed point in time. Unlike the `SERIALIZABLE` level, `SNAPSHOT ISOLATION` permits the write skew anomaly, but in cases where write skew conditions are unlikely, this isolation level can be highly performant.  
```

Comment:
For future reference, I did have an entry for write skew in my glossary which you are welcome to use.